Sources/MapboxNavigation/InstructionPresenter.swift

```diff
@@ -179,7 +201,8 @@ class InstructionPresenter {
             attachment.image = image
         } else {
             let view = GenericRouteShield(pointSize: dataSource.font.pointSize, text: text)
-            view.foregroundColor = dataSource.textColor
+            view.foregroundColor = GenericRouteShield.appearance().foregroundColor
+            view.borderWidth = GenericRouteShield.appearance().borderWidth
```

Comment:
@avi-c, it's been a while since this logic was modified, but do you remember why you've decided to use `dataSource.textColor` instead of `GenericRouteShield`s appearance. Maybe I'm missing some edge case here.

Reply:
#2280 was intended to fix #2161, but #2977 ended up actually fixing that issue when switching between day and night. Not sure if the changes in #2280 are still required.

Reply:
One thing to watch out for is that CarPlay always uses the night style, even during the daytime when the phone uses the day style.

Reply:
Behavior was changed: user will have to provide two different appearances for iOS and CarPlay, Navigation SDK by default also provides colors for different interface styles.

Reply:
Your first comment got buried under my email notifications. 

I can't recall exactly why I chose the data source over the straight appearance proxy but it might have been that I was using the shield within another view container and needed it to defer to the datasource to find the right color.